src/secret.rs

```diff
@@ -46,7 +46,31 @@ macro_rules! impl_display_secret {
                 engine.input(&self.secret_bytes());
                 let hash = sha256::Hash::from_engine(engine);
 
-                f.debug_tuple(stringify!($thing)).field(&format_args!("#{:016x}", hash)).finish()
+                // hex-conservative does not support truncation and this is the correct behaviour if
```

Comment:
We don't use `hex-conservative` at all, so I'm not sure why this reference is here. But I would argue we do support it - if you take the subslice. :)

Reply:
I thought secp depends on hashes which depends on hex which does the formatting.

Reply:
Oh, I haven't noticed this is in a branch that has the `hashes` feature on. We can just use `hex` then if it's reexported. (Though this somewhat goes against our idea of having `hex` in hashes as optional.)

Reply:
So my code comment stands, right?  `hex-conservative` does not currently support truncating.

Reply:
Depends on point of view. We do not support truncating via formatter and we have no hack to get it for types that implemented hex formatting without doing any effort to support truncating.

Reply:
IOW "yes, it supports truncating hashes using the formatter". This was an explicit feature request and one of the very first things we put into hex-conservative.

Reply:
In that case I must've misremembered something. I do remember we were trying to solve it though.

Reply:
Well, we do it because to format hashes we pass through to the formatter for slices, which matches what you said.

Though somehow we also do truncation for reversed hashes (I just checked, we do) so maybe we do something more clever..